src/linear_algebra/basic.lean

```diff
@@ -1572,6 +1572,49 @@ end field
 
 end linear_equiv
 
+namespace submodule
+
+variables [comm_ring R] [add_comm_group M] [add_comm_group M₂] [module R M] [module R M₂]
+variables (p : submodule R M) (q : submodule R M₂)
+
+lemma comap_le_comap_smul (f : M →ₗ[R] M₂) (c : R) :
+  comap f q ≤ comap (c • f) q :=
+begin
+  rw le_def',
+  intros m h,
+  change c • (f m) ∈ q,
+  change f m ∈ q at h,
+  apply submodule.smul _ _ h,
+end
+
+lemma inf_comap_le_comap_add (f₁ f₂ : M →ₗ[R] M₂) :
+  comap f₁ q ⊓ comap f₂ q ≤ comap (f₁ + f₂) q :=
+begin
+  rw le_def',
+  intros m h,
+  change f₁ m + f₂ m ∈ q,
+  change f₁ m ∈ q ∧ f₂ m ∈ q at h,
+  apply submodule.add _ h.1 h.2,
+end
+
+/-- Given modules `M`, `N` over a commutative ring, together with submodules `p ⊆ M`, `q ⊆ N`, the
+set of maps $\{f ∈ Hom(M, N) | f(p) ⊆ q \}$ is a submodule of `Hom(M, N)`. -/
+def comap_linear_maps : submodule R (M →ₗ[R] M₂) :=
```

Comment:
Could we have variables names in the comment match those in the statement? (I'd prefer to change the ones in the statement.)

Reply:
Done.

I do prefer `N` instead of `M₂` but the idiom in this file seems to be to use `M₂` so I decided to fall in line with that, and change the comment rather than the statement.